modules/stochastic_tools/src/surrogates/GaussianProcessTrainer.C

```diff
@@ -67,6 +70,9 @@ GaussianProcessTrainer::GaussianProcessTrainer(const InputParameters & parameter
     _standardize_data(getParam<bool>("standardize_data")),
     _do_tuning(isParamValid("tune_parameters")),
     _tuning_algorithm(getParam<MooseEnum>("tuning_algorithm")),
+    _iter_ADAM(getParam<unsigned int>("iter_ADAM")),
+    _batch_size(isParamValid("batch_size") ? &getParam<unsigned int>("batch_size") : nullptr),
```

Comment:
Do we need this pointer treatment? I'd recommend using a regular unsigned int and just add a batch size of 0 by default. Seems to me it will have the same effect.

Reply:
Also you can use` isParamSetByUser()` for `isParamValid()` if this causes issues.